src/index/IndexImpl.Text.cpp

```diff
@@ -371,13 +344,12 @@ void IndexImpl::wordsFromDocsFileEntitiesFromWordsFile(
     const LocaleManager& localeManager, T processLine) const {
   // Initialize DocsFileParser and WordsFileParser and the respective
   // iterators to parse in parallel
-  auto docsFileParser = DocsFileParser{docsFile, localeManager};
-  auto docsFileIterator = docsFileParser.begin();
   auto wordsFileParser = WordsFileParser{wordsFile, localeManager};
   auto wordsFileIterator = wordsFileParser.begin();
-  // Get the respective lines
-  WordsFileLine currentWordsFileLine = *wordsFileIterator;
-  DocsFileLine currentDocsFileLine = *docsFileIterator;
+  AD_CORRECTNESS_CHECK(wordsFileParser.begin() != wordsFileParser.end(),
+                       "When adding entities from wordsfile the given "
+                       "wordsfile can't be an empty file.");
```

Comment:
```suggestion
  AD_CORRECTNESS_CHECK(wordsFileIterator.begin() != wordsFileParser.end(),
                       "When adding entities from wordsfile the given "
                       "wordsfile can't be an empty file.");
```
(if WordsFilePraser is an inputRange like a generator, then calling begin() multiple times can be a problem).